log4j-api/src/main/java/org/apache/logging/log4j/spi/DefaultThreadContextMap.java

```diff
@@ -196,6 +193,9 @@ public boolean equals(final Object obj) {
         if (obj == null) {
             return false;
         }
+        if (obj instanceof ReadOnlyStringMap) {
+            return ReadOnlyStringMapUtil.equals(this, (ReadOnlyStringMap) obj);
+        }
```

Comment:
`DefaultThreadContextMap` is primarily an implementation of `ThreadContextMap`. I am not sure if the fact it implements `ReadOnlyStringMap` is ever used.